### PR TITLE
[go1.15] Update to go1.15.5

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind feature
/area dependency

**What this PR does / why we need it**:

- Update to go1.15.5
- build: Update to k/repo-infra@v0.1.3 (supports go1.15.5)

Tracking issue: https://github.com/kubernetes/release/issues/1651
Slack thread: https://kubernetes.slack.com/archives/C2C40FMNF/p1605129514273400
cc: @kubernetes/release-engineering 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

/hold this still needs a go-runner update

**Does this PR introduce a user-facing change?**:

```release-note
- Kubernetes is now built using go1.15.5
- build: Update to k/repo-infra@v0.1.2 (supports go1.15.5)
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by justaugustus** [comment]
```console
ERROR: /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/external/io_bazel_rules_go/BUILD.bazel:29:1: GoStdlib external/io_bazel_rules_go/stdlib_/pkg failed (Exit 34). Note: Remote connection/protocol failed with: execution failed catastrophically UNAVAILABLE: HTTP/2 error code: NO_ERROR 
```

:this-is-fine:
/test pull-kubernetes-bazel-build

**Review by saschagrunert — APPROVED** [review]
/lgtm

File: build/dependencies.yaml [hunk 1]
```diff
@@ -99,22 +99,22 @@ dependencies:
...
       match: GOLANG_VERSION=\d+.\d+(alpha|beta|rc)?\.?\d+
     - path: staging/publishing/rules.yaml
       match: 'default-go-version\: \d+.\d+(alpha|beta|rc)?\.?(\d+)?'
 
   - name: "k8s.gcr.io/kube-cross: dependents"
-    version: v1.15.2-1
+    version: v1.15.5-1
     refPaths:
     - path: build/build-image/cross/VERSION
-    - path: test/images/sample-apiserver/Makefile
```
Comment (liggitt) [inline build/dependencies.yaml:116]: curious why this got dropped

**Review by liggitt — COMMENTED** [review]
one question, lgtm otherwise

File: build/dependencies.yaml (hunk 1 above)
Comment (justaugustus) [reply build/dependencies.yaml:116]: @liggitt -- the sample apiserver uses a different image now and the reference no longer exists: https://github.com/kubernetes/kubernetes/blob/0c973409051faa8926971cfb05b967a2d88c5ca8/test/images/sample-apiserver/BASEIMAGE#L1

Comment (liggitt) [reply build/dependencies.yaml:116]: ack

Comment (liggitt) [reply build/dependencies.yaml:116]: actually, is that true? I still see `k8s.gcr.io/build-image/kube-cross:v1.15.2-1` at https://github.com/kubernetes/kubernetes/blob/master/test/images/sample-apiserver/Makefile#L27

**Comment by MonzElmasry** [comment]
@liggitt is this one approved?

**Comment by liggitt** [comment]
> @liggitt is this one approved?

https://github.com/kubernetes/kubernetes/pull/95776#discussion_r525391271 is still outstanding

**Comment by justaugustus** [comment]
/hold cancel

Fixed https://github.com/kubernetes/kubernetes/pull/95776#discussion_r525391271 and added in the go-runner go1.15.5 image (`go-runner:buster-v2.2.2`) from https://github.com/kubernetes/release/pull/1780.

**Pull request merged**